services/inventory/nodes.go

```diff
@@ -234,3 +227,54 @@ func (s *NodesService) AddRemoteRDSNode(ctx context.Context, req *inventorypb.Ad
 
 	return invNode.(*inventorypb.RemoteRDSNode), nil
 }
+
+// Remove removes Node without any Agents and Services.
+// Removes Node with the Agents and Services if force == true.
+// Returns an error if force == false and Node has Agents or Services.
+func (s *NodesService) Remove(ctx context.Context, id string, force bool) error {
+	pmmAgentIDs := make(map[string]struct{})
+
+	if e := s.db.InTransaction(func(tx *reform.TX) error {
+		mode := models.RemoveRestrict
+		if force {
+			mode = models.RemoveCascade
+
+			allAgents, err := models.FindPMMAgentsRunningOnNode(tx.Querier, id)
+			if err != nil {
+				return errors.WithStack(err)
+			}
+
+			agents, err := models.FindAgents(tx.Querier, models.AgentFilters{NodeID: id})
+			if err != nil {
+				return errors.WithStack(err)
+			}
+			allAgents = append(allAgents, agents...)
+
+			agents, err = models.FindPMMAgentsForServicesOnNode(tx.Querier, id)
```

Comment:
These  pmm-agents doesn't have `PMMAgentID` field, we should use just `ID` for them.